discovery/manager_test.go

```diff
@@ -949,6 +950,82 @@ scrape_configs:
 	}
 }
 
+func TestGaugeFailedConfigs(t *testing.T) {
+	cfgOneText := `
+scrape_configs:
+- job_name: prometheus
+  consul_sd_configs:
+  - server: "foo:8500"
+    tls_config:
+      cert_file: "/tmp/not_existed"
+  - server: "bar:8500"
+    tls_config:
+      cert_file: "/tmp/not_existed"
+  - server: "foo2:8500"
+    tls_config:
+      cert_file: "/tmp/not_existed"
+`
+	cfgOne := &config.Config{}
+
+	if err := yaml.UnmarshalStrict([]byte(cfgOneText), cfgOne); err != nil {
+		t.Fatalf("Unable to load YAML config cfgOne: %s", err)
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	discoveryManager := NewManager(ctx, log.NewNopLogger())
+	discoveryManager.updatert = 100 * time.Millisecond
+	go discoveryManager.Run()
+
+	c := make(map[string]sd_config.ServiceDiscoveryConfig)
+	for _, v := range cfgOne.ScrapeConfigs {
+		c[v.JobName] = v.ServiceDiscoveryConfig
+	}
+
+	discoveryManager.ApplyConfig(c)
+	<-discoveryManager.SyncCh()
+
+	metricOne := &dto.Metric{}
+	if counter, err := failedConfigs.GetMetricWithLabelValues(discoveryManager.name); err != nil {
+		t.Fatalf("%s", err)
+	} else {
+		counter.Write(metricOne)
+	}
+
+	if metricOne.GetGauge().GetValue() != 3 {
+		t.Fatalf("Expected to have 3 failed configs")
```

Comment:
Would be good to include the received count as well in this message: "Expected 3 failed configs, got X."